lib/luacrypto.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+ * SPDX-License-Identifier: MIT OR GPL-2.0-only
+ */
+
+#ifndef _LUACRYPTO_H
+#define _LUACRYPTO_H
+
+#include <linux/err.h>
+#include <lua.h>
+#include <lauxlib.h>
+#include <lunatik.h>
+
+/**
+ * LUACRYPTO_NEW - Macro to define a common constructor for crypto transform objects.
+ * @name: The base name of the crypto module (e.g., aead, skcipher, shash).
+ * @T: The C struct type of the crypto transform (e.g., struct crypto_aead).
+ * @alloc: The kernel crypto allocation function (e.g., crypto_alloc_aead).
+ * @class: The lunatik_class_t instance for this object type.
+ * @private: The expression to assign to object->private (e.g., tfm or sdesc).
+ * @define_priv: Optional extra allocation and assignment logic (e.g., for shash_desc).
+ *
+ * This macro generates the `luacrypto_<_name>_new` function, which acts as the
+ * constructor for Lua crypto objects. It handles the common steps of checking
+ * the algorithm name, allocating the Lunatik object, allocating the kernel
+ * crypto transform, handling errors, and assigning the private data.
+ */
+#define LUACRYPTO_NEW_CUSTOM(name, T, alloc, class, define_priv, ...)					\
```

Comment:
we can use the same strategy from here (a GCC extension) to get rid of `, `:
https://github.com/luainkernel/lunatik/blob/master/lunatik.h#L50C17-L50C21